python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -225,15 +223,24 @@ cpdef read_parquet(filepaths_or_buffers, columns=None, row_groups=None,
     column_names = [x.decode() for x in c_out_table.metadata.column_names]
 
     # Access the Parquet user_data json to find the index
-    index_col = ''
+    index_col = None
     cdef map[string, string] user_data = c_out_table.metadata.user_data
     json_str = user_data[b'pandas'].decode('utf-8')
     meta = None
     if json_str != "":
         meta = json.loads(json_str)
         if 'index_columns' in meta and len(meta['index_columns']) > 0:
-            index_col = meta['index_columns'][0]
-
+            index_col = meta['index_columns']
+            if isinstance(index_col[0], dict) and \
+                    index_col[0]['kind'] == 'range':
+                is_range_index = True
+            else:
+                is_range_index = False
+                index_col_names = OrderedDict()
+                for idx_col in index_col:
+                    for c in meta['columns']:
+                        if c['field_name'] == idx_col:
+                            index_col_names[idx_col] = c['name']
```

Comment:
Should be able to use a comprehension for this.

Reply:
```
index_col_names = OrderedDict(
                    (idx_col, c['name']) for idx_col in index_col for c in meta['columns'] if c['field_name'] == idx_col
                )
```

Using this seems to be not supported by cython compiler.

```
Error compiling Cython file:
------------------------------------------------------------
...
                    index_col[0]['kind'] == 'range':
                is_range_index = True
            else:
                is_range_index = False
                index_col_names = OrderedDict(
                    (idx_col, c['name']) for idx_col in index_col for c in meta['columns'] if c['field_name'] == idx_col
                    ^
------------------------------------------------------------

cudf/_lib/parquet.pyx:240:21: Compiler crash in CreateClosureClasses

ModuleNode.body = StatListNode(parquet.pyx:5:0)
StatListNode.stats[28] = StatListNode(parquet.pyx:171:6)
StatListNode.stats[0] = CFuncDefNode(parquet.pyx:171:6,
    args = [...]/7,
    doc = '\n    Cython function to call into libcudf API, see `read_parquet`.\n\n    See Also\n    --------\n    cudf.io.parquet.read_parquet\n    cudf.io.parquet.to_parquet\n    ',
    modifiers = [...]/0,
    needs_closure = True,
    overridable = 1,
    visibility = 'private')
CFuncDefNode.body = StatListNode(parquet.pyx:174:4,
    is_terminator = True)
StatListNode.stats[15] = IfStatNode(parquet.pyx:230:4)
IfStatNode.if_clauses[0] = IfClauseNode(parquet.pyx:230:7)
IfClauseNode.body = StatListNode(parquet.pyx:231:8)
StatListNode.stats[1] = IfStatNode(parquet.pyx:232:8)
IfStatNode.if_clauses[0] = IfClauseNode(parquet.pyx:232:11)
IfClauseNode.body = StatListNode(parquet.pyx:233:12)
```